easybuild/easyconfigs/u/UQTk/UQTk-3.1.0_fix-blas-lapack.patch

```diff
@@ -0,0 +1,1293 @@
+-target_link_libraries (SurfRxnNISP_MC.x m blas ${BLAS_LIBRARIES})
++target_link_libraries (SurfRxnNISP_MC.x m $ENV{LIBLAPACK_MT} ${LAPACK_LIBRARIES})
++target_link_libraries (SurfRxnNISP_MC.x m $ENV{LIBBLAS} ${BLAS_LIBRARIES})
+
+ target_link_libraries (SurfRxnISP.x  uqtkpce uqtkmcmc uqtkquad uqtktools uqtkarray uqtkxmlutils)
+ target_link_libraries (SurfRxnISP.x sundials_cvode sundials_nvecserial sundials_sunlinsoldense sundials_sunmatrixdense)
+ target_link_libraries (SurfRxnISP.x  depdsfmt deplbfgs depslatec depfigtree depann)
+-target_link_libraries (SurfRxnISP.x m lapack ${LAPACK_LIBRARIES})
+-target_link_libraries (SurfRxnISP.x m blas ${BLAS_LIBRARIES})
++target_link_libraries (SurfRxnISP.x m $ENV{LIBLAPACK_MT} ${LAPACK_LIBRARIES})
++target_link_libraries (SurfRxnISP.x m $ENV{LIBBLAS} ${BLAS_LIBRARIES})
+
+ target_link_libraries (SurfRxnDet.x  uqtkpce uqtkmcmc uqtkquad uqtktools uqtkarray uqtkxmlutils)
+ target_link_libraries (SurfRxnDet.x sundials_cvode sundials_nvecserial sundials_sunlinsoldense sundials_sunmatrixdense)
+ target_link_libraries (SurfRxnDet.x  depdsfmt deplbfgs depslatec depfigtree depann)
+-target_link_libraries (SurfRxnDet.x m lapack ${LAPACK_LIBRARIES})
+-target_link_libraries (SurfRxnDet.x m blas ${BLAS_LIBRARIES})
++target_link_libraries (SurfRxnDet.x m $ENV{LIBLAPACK_MT} ${LAPACK_LIBRARIES})
++target_link_libraries (SurfRxnDet.x m $ENV{LIBBLAS} ${BLAS_LIBRARIES})
+
+ # Link fortran libraries
+ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+diff -ru UQTk-3.1.0.orig/examples/tmcmc_bimodal/CMakeLists.txt UQTk-3.1.0/examples/tmcmc_bimodal/CMakeLists.txt
+--- UQTk-3.1.0.orig/examples/tmcmc_bimodal/CMakeLists.txt	2020-03-10 22:59:24.000000000 +0100
++++ UQTk-3.1.0/examples/tmcmc_bimodal/CMakeLists.txt	2020-03-28 18:18:38.287592625 +0100
+@@ -9,7 +9,7 @@
+   README
+   )
+
+-link_directories(${CMAKE_SUNDIALS_DIR}/lib)
++link_directories(${CMAKE_SUNDIALS_DIR}/lib64)
+
+ add_executable (bimodal.x bimodal.cpp)
+ add_executable (tmcmc_bimodal.x tmcmc_bimodal.cpp)
+@@ -21,8 +21,8 @@
+ target_link_libraries (tmcmc_bimodal.x sundials_sunlinsoldense)
+ target_link_libraries (tmcmc_bimodal.x sundials_sunmatrixdense)
+ target_link_libraries (tmcmc_bimodal.x depslatec)
+-target_link_libraries (tmcmc_bimodal.x m lapack)
+-target_link_libraries (tmcmc_bimodal.x m blas)
++target_link_libraries (tmcmc_bimodal.x m $ENV{LIBLAPACK_MT})
++target_link_libraries (tmcmc_bimodal.x m $ENV{LIBBLAS})
+ target_link_libraries (tmcmc_bimodal.x deplbfgs  )
+
+
+diff -ru UQTk-3.1.0.orig/PyUQTk/bcs/bcs_ext.py UQTk-3.1.0/PyUQTk/bcs/bcs_ext.py
+--- UQTk-3.1.0.orig/PyUQTk/bcs/bcs_ext.py	2020-03-10 22:59:24.000000000 +0100
++++ UQTk-3.1.0/PyUQTk/bcs/bcs_ext.py	2020-03-31 14:53:05.277127716 +0200
+@@ -30,10 +30,23 @@
+
+ import numpy as np
+ import matplotlib.pyplot as mpl
+-import uqtkarray as uqtkarray
+-import pce as uqtkpce
+-import tools as uqtktools
+-from uqtkarray import uqtk2numpy, numpy2uqtk
++try:
++	import PyUQTk.uqtkarray as uqtkarray
```

Comment:
@Darkless012 Hmm, this is strage... Is this the only script you had to fix like this?

Reply:
This tell me we should be adding `%(installdir)s/PyUQTK` to `$PYTHONPATH`, since the modules are expected to be directly importable... See also remarks in `UQTk` easyconfig file w.r..t `modextrapaths` and `sanity_check_commands`

So you should remove this part of the patch (the whole block for `UQTk-3.1.0/PyUQTk/bcs/bcs_ext.py`), and update the SHA256 checksum for the patch file.

Reply:
Tests are loading the scripts including their folder before install. Which result in having everything in path. So tests pass, but importing this fails because on PYTHONPATH is parent folder. 
This looks like only script requiring other modules.
see:
https://github.com/sandialabs/UQTk/blob/master/PyUQTk/__init__.py
https://github.com/sandialabs/UQTk/blob/master/PyUQTk/pytests/PyModTest.py
They pretty much did the same here:
https://github.com/sandialabs/UQTk/blob/master/PyUQTk/PyPCE/adaptation_tools.py

I think that those are relics from Python2-only and the way it handled relative/absolute imports.

Reply:
OK, thanks for clarifying